geoh5py/workspace/workspace.py

```diff
@@ -378,42 +375,48 @@ def create_object_or_group(
         return None
 
     @property
-    def data(self) -> List["data.Data"]:
+    def data(self) -> List["Entity"]:
```

Comment:
why using the more generic type `Entity` when it is returning `Data`?

Reply:
Same for all the comments below, I am now getting an error from mypy regarding the mix of types
![image](https://user-images.githubusercontent.com/55204635/118292807-dd3bbc00-b48d-11eb-8b29-54558008bcad.png)

Any idea?

Reply:
found this https://stackoverflow.com/questions/56738485/why-do-i-get-a-warning-when-concatenating-lists-of-mixed-types-in-pycharm

Looks like I would need to avoid adding the lists....

Reply:
I would think you just need to cast when adding them:
`my_base_type_list = cast(List[base_type],my_derived_type1_list) + cast(List[base_type],my_derived_type2_list)`

Reply:
Done